src/CpmsClient/Service/ApiService.php

```diff
@@ -104,14 +106,7 @@ protected function processRequest($endPointAlias, $scope, $method, $params = nul
 
                 $this->getOptions()->setHeaders($headers);
 
-                if (empty($data['customer_reference'])) {
-                    $data['customer_reference'] = $this->options->getCustomerReference();
-                }
-
-                if (empty($data['user_id'])) {
-                    $data['user_id'] = $this->options->getUserId();
-                }
```

Comment:
Was that code not needed?

Reply:
not sure, but there seems to be no variable $data, plus data['user_id'] gives a warning "write access unused"

Reply:
re-added code, it needed a different parameter name $params.